### DISCUSSION TITLE
Project: ✅ Wagtail.org website accessibility

### DISCUSSION THREAD

**Discussion by thibaudcolas** [wagtail/outreachy#3] [Ideas]
Please use this Discussions thread to ask anything and everything relating to our [Wagtail.org website accessibility](https://github.com/wagtail/outreachy/blob/main/project-ideas.md#wagtailorg-website-accessibility) project idea.

Don’t hesitate to answer other people’s questions and mark this as a contribution on the Outreachy website if you’re particularly happy with your answer.

**Comment by thibaudcolas** [comment]
Q: I saw a project called "Wagtail.org accessibility" on Outreachy. What’s the difference? Are they the same?

**Reply by thibaudcolas** [reply]
I’m glad you asked @thibaudcolas! Yes, this is the one and only "wagtail.org" project. I just changed the name to hopefully clarify it’s about a specific website.

**Comment by MugoBrian** [comment]
Hello @thibaudcolas . While trying to install wagtail.org locally , I am getting the following error while running the command make setup, I tried with and without sudo.
Screenshot of the error.
![Screenshot from 2023-10-06 18-10-21](https://github.com/wagtail/outreachy/assets/49617104/db223931-f9e9-477f-ae87-124a41ef6a98)

Kindly assist me.

**Reply by thibaudcolas** [reply]
Just to let you know I’ve been investigating, and am still at it unfortunately. I’ll get back to you.

**Reply by MugoBrian** [reply]
Noted with thanks.

**Reply by thibaudcolas** [reply]
@MugoBrian could you give this another go, making sure you use the latest version of `wagtail.org`? I’ve integrated changes that should make the setup more robust. 

If this still fails, I’ll add even simpler setup instructions that don’t rely on Docker.

**Reply by MugoBrian** [reply]
I will . I might say Ubuntu 22.0.4 has some issues when it comes to permissions, since it can't run the `make` command without `sudo`. Perhaps it include be included in the docs or README as a small note, if a user encounters `permission denied error` to run the command with `sudo`

**Reply by thibaudcolas** [reply]
@MugoBrian I do not see a fundamental reason why `sudo` would be needed – are you sure that’s not a specific issue with your setup?

**Reply by MugoBrian** [reply]
Yes. I believe it's a specific issue. As I had mentioned earlier Ubuntu 22.0.4 has some issues when it comes to permission which forces me to add sudo.

**Reply by MugoBrian** [reply]
Hello @thibaudcolas, I tried installing it again and got new errors.

![Screenshot from 2023-10-16 10-50-11](https://github.com/wagtail/outreachy/assets/49617104/ca93acec-b988-4591-b73d-119996c1b657)

**Reply by praiztech** [reply]
@thibaudcolas I find that I'm unable to use the 'make' command to set up wagtail.org for development.
I think this is similar to the issue MugoBrian is experiencing.
When I tried to run the 'make setup' command, I get a "command 'make' not found" message.
I'm running Ubuntu for Wsl via Windows Terminal with Docker Wsl integration.
I've attached a screenshot of my terminal.

![My windows terminal showing the process of cloning the wagtail.org repo and my attempt at setting it up for development.](https://github.com/wagtail/outreachy/assets/66212418/e749b10e-944a-40b5-ac76-93bf9b232350)

**Reply by MugoBrian** [reply]
@praiztech .Kindly confirm you have make installed in your system. 
Since the terminal is suggesting you install make using
the command `sudo apt install make` 
Alternatively you can use Gitpod to set up the codebase.

**Reply by praiztech** [reply]
@MugoBrian I noticed you were experiencing issues with the 'make setup' command as well. So, I thought I'd seek clarifications before moving ahead.

If you don't mind, what are the issues you're experiencing? Is it different from mine? I'm new to Linux OS.

**Reply by MugoBrian** [reply]
In the updated DockerFile , there's a step failing which includes installing and creating a Python virtualenv.
Ooh, Linux can be quite challenging  especially the commands.

@thibaudcolas , I would suggest if in the Readme there was a statement informing the user to have make installed on their system. What is your take on that also @praiztech ?

**Reply by praiztech** [reply]
Yes. That's a good idea @MugoBrian

**Reply by praiztech** [reply]
As this issue you're experiencing seems to show up on Ubuntu, isn't it likely that I'd run into the same issue since I'm running Ubuntu as well @MugoBrian?

**Reply by MugoBrian** [reply]
I believe so.After installing make , you can try setting up and see if `make setup` will run successfully. 
If it will work , I'll be so happy since you managed to progress and you'll be able to assist me too.

**Reply by praiztech** [reply]
All right. I'll let you know. Thanks.

**Reply by MugoBrian** [reply]
You welcome

**Reply by praiztech** [reply]
@MugoBrian I have finally been able to set up the wagtail.org repo on my local machine.
Since I'm new to Linux, I'll just list the sequence of steps I took.
I read that it's best to upgrade all outdated packages before installing 'make' so I did that first. 
Then, I installed 'make'. I restarted my PC, docker and Ubuntu, and ran the 'make setup' command. 
After some time running, I got a '403 connection to debian server failed' error. I searched the web and found suggestions to reset my router. 
I did just that, restarted docker and Ubuntu, and ran the 'make setup' command again. The error went away and the setup ran to completion.
Running the other commands worked smoothly once setup was complete.
Hopefully, this can help.

**Reply by nishantn31** [reply]
Cool! @MugoBrian 
Let us know if you're stuck in the further process. We'll try our best to help you out!

**Reply by MugoBrian** [reply]
Whoa! I am proud of you @praiztech .I am still facing challenges. This time round is a new error. Installation of Poetry is failing

**Reply by praiztech** [reply]
Thanks @MugoBrian. 

I don't really know how linux works but is it possible to maybe uninstall and reinstall the OS? I had to do a clean install of Windows 10 64-bit before I could install Docker.

**Reply by MugoBrian** [reply]
Doing a clean install is not feasible for me. I'll do more research on my errors.For now I'll stick to Gitpod setup.

**Reply by MugoBrian** [reply]
@praiztech and @thibaudcolas I have successfully managed to setup the codebase locally using Docker. Thank you.

**Reply by praiztech** [reply]
Finally!

**Comment by khushishikhu** [comment]
![Screenshot 2023-10-09 224626](https://github.com/wagtail/outreachy/assets/65439761/cdf7fc33-6b28-40c4-a6fd-9cf1d03a00d2)
I tried to setup wagtail and did all the steps but it seems to show me this page on localhost. Please help me with this issue

**Reply by thibaudcolas** [reply]
@khushishikhu this looks alright to me, though the docs could be clearer! I’m glad you reached this step :) You essentially have an empty site here. From there, you need to log into the CMS and create test content. Most likely replace this placeholder page with a real homepage, and then expand from there.

**Reply by khushishikhu** [reply]
@thibaudcolas could you please guide me a bit about how to log into CMS and move ahead. I can't find any link which has this step clearly mentioned

**Reply by MugoBrian** [reply]
Perhaps you could follow this tutorial on
[Your first wagtail site](https://docs.wagtail.org/en/latest/getting_started/tutorial.html)

**Reply by thibaudcolas** [reply]
@khushishikhu I would expect this is the kind of thing you can figure out on your own. If you don’t have Django experience, I’d recommend going through their tutorial or our [Your first Wagtail site](https://docs.wagtail.org/en/latest/getting_started/tutorial.html) as mentioned by @MugoBrian before trying to contribute to this project.

**Comment by khushishikhu** [comment]
![Screenshot 2023-10-10 014742](https://github.com/wagtail/outreachy/assets/65439761/bda9e0c1-0cad-4b45-a72e-e385db1d749a)

@thibaudcolas I tried logging into the CMS but ended up with this error, the moment I clicked login. Can you please help me to resolve it

**Reply by MugoBrian** [reply]
Could you highlight the steps you took, I try to reproduce it on my end?
Running migrations and creating a superuser then logging in with the superuser details created, helped me.

**Reply by khushishikhu** [reply]
@MugoBrian I just ran "make start", "make runserver", "make migrate" and then made the supersuser. Then i went to loggin page and when I tried to login , it gave me this page

**Reply by Chidera6** [reply]
@khushishikhu  I had this error as well when  setting up wagtail for development. That is when I tried to log in to the admin dashboard. I discovered I missed a step, which is to createcache table. I wrote [this](https://dev.to/chidera/setting-up-wagtail-for-development-on-windows-a-step-by-step-guide-j7p) blog post which captures all the steps I took, maybe it will help you. Aside that, I also discovered I did not start `storybook` in  a separate terminal as explained in these codes below from the official website.
 ```export DJANGO_SETTINGS_MODULE=wagtail.test.settings_ui
# Assumes the current environment contains a valid installation of Wagtail for local development.
./wagtail/test/manage.py migrate
./wagtail/test/manage.py createcachetable
./wagtail/test/manage.py runserver 0:8000
# In a separate terminal:
npm run storybook
```

**Reply by khushishikhu** [reply]
@Chidera6 do we need to make the test directory ourself?

**Reply by Chidera6** [reply]
@khushishikhu No, It's already there from the source code. You will see it inside the wagtail directory, the path looks like this `wagtail/wagtail/test`.

**Reply by khushishikhu** [reply]
@Chidera6 I am sorry, I am not able to identify that. Could you please send me the link of that directory here. It will be very helpful for me

**Reply by Chidera6** [reply]
Hi @khushishikhu. I installed it locally, so there is no link I can send you. Do you mind us getting on a huddle? Maybe we can figure it out that way.

**Reply by khushishikhu** [reply]
@Chidera6 yes sure, it will be very nice of you. Let me know when

**Reply by Chidera6** [reply]
@khushishikhu Now, if that's okay by you. I sent you a message on Slack.

**Reply by thibaudcolas** [reply]
@Chidera6 the instructions you share are about getting Wagtail itself set up for local development. If @khushishikhu is asking here I believe it’s [wagtail.org](https://github.com/wagtail/wagtail.org) she wants help setting up?

**Reply by thibaudcolas** [reply]
@khushishikhu from what you shared it looks like you didn’t run the `make setup` step? Please make sure to follow the instructions carefully, that’s meant to be the first step in the setup, not `make start`.

**Reply by Chidera6** [reply]
@thibaudcolas Yes that's what she is asking for. I realized I mixed it up during the video call I had with her.

**Reply by khushishikhu** [reply]
@thibaudcolas I did run `make setup` and after that I run `make start` the. I created the superuser but it still gave me this error

**Reply by MugoBrian** [reply]
@khushishikhu ,did you try to do a fresh install of the wagtail.org. Sometimes that's what I do if no solution is working. I would delete the repository and start over again , it helps sometimes.

Have you tried setting it up using Gitpod?

**Comment by nouman17610** [comment]
@thibaudcolas I have a quick question. When would you like us to record contributions at outreach? with every task or only the main issue?

**Reply by thibaudcolas** [reply]
@nouman17610 please keep this for questions about the project. I’ve answered [on the Helpdesk thread](https://github.com/wagtail/outreachy/discussions/30#discussioncomment-7273016).

**Comment by thibaudcolas** [comment]
Q by @Amarajah on Slack:

I'm working on contributing to the [Wagtail.org](http://wagtail.org/) website accessibility project, one of the contribution requirements is:

> Find an accessibility issue on the live [wagtail.org](http://wagtail.org/) website, and report it on the project’s issue tracker, with appropriate details (title, description, screenshot if applicable, why you think this is an accessibility issue).

I went through the site and was unable to find anything I could report, but on the Wagtail GitHub repo I found some issues regarding the live site which were still open.

Am I allowed to work on one of those issues? Or is it mandatory I find my own errors?
If we're allowed to work on those already reported issues, how many should we contribute to? Is one enough?

**Reply by thibaudcolas** [reply]
**Yes, you are allowed to work on those pre-existing issues**. It’s not mandatory to find your own errors.

As part of working on those pre-existing issues, I’d still recommend taking a moment to check whether you can actually spot the issue – confirming whether the report is clear enough, and whether it’s still up-to-date (if the issue is old). Taking the time to comment on the issue with a summary of your findings is a great contribution. Something along the lines of "I’m going to work on this. I can confirm this is still an issue – I’ve tested this on \<page you tested it on, with link\>, with the following steps / tool." Perhaps a screenshot if you think it helps.

---

> If we're allowed to work on those already reported issues, how many should we contribute to? Is one enough?

**Yes, one is enough**. I’d say zero can also be enough. We have those tasks on our checklist for three reasons, but there are other ways to make it work:

1. So you can check whether the project is a good fit for you. The actual internship will be full of tasks like this – going through it one time is a good sense-check that this will be interesting for you, and that you have the right skillset.
2. So mentors can check whether you’re a good fit for the project. You can prove yourself in other ways but doing something that’s similar to what the project involves is definitely a great approach.
3. Because those are useful contributions! That one is a bit obvious but I guess worth saying – those issues are really important for us to identify and fix.

One is plenty enough if it fits the bill on those three criteria. So if you did only one, strategically I’d recommend picking an issue that gives you plenty of opportunity to test/demonstrate your abilities. From my perspective as a mentor, I’d much rather review one thing you’ve put serious thought into, than 2/3/4/5 that were all very basic. Quality over quantity.

**Reply by praiztech** [reply]
Thank you

**Reply by nishantn31** [reply]
Thank you @thibaudcolas for discussing it in detail. I had recently successfully raised a PR for the [issue](https://github.com/wagtail/wagtail.org/issues/351) after checking it's existence in live site and now studying project further for accessibility improvements. Also updating my checklist with every task I'm completing.

**Reply by thibaudcolas** [reply]
Nice one @nishantn31, I’ll try to review it soon.

**Reply by nishantn31** [reply]
Thank you! @thibaudcolas

**Comment by MugoBrian** [comment]
Hello @thibaudcolas. I believe you are doing good. I have two questions, in particular, I would love help with. 
1. How do I live preview pages such as /roadmap or /packages, in the wagtail.org code base? I would like to see the changes I made affecting those pages.
2. How do I solve these errors?
![Screenshot from 2023-10-22 21-32-57](https://github.com/wagtail/outreachy/assets/49617104/cf975030-6f48-4a29-b7ea-da57dad78d87)
![Screenshot from 2023-10-22 21-32-08](https://github.com/wagtail/outreachy/assets/49617104/69200505-0a63-43e1-bc81-3c9638209830)

I am using Gitpod to run the wagtail.org codebase.

**Reply by oly-baby** [reply]
For the first question, I think the packages page is an API, it's django-packages Api that was used.

For the second question, I will love to know the answer too

**Reply by MugoBrian** [reply]
Thank you @oly-baby .So what would be the path to that page?

**Reply by oly-baby** [reply]
You have to check Django package repo

**Reply by thibaudcolas** [reply]
@MugoBrian could you open an issue about this? I don’t think you have any need to use this view for the time being though.

**Reply by MugoBrian** [reply]
Ok. I will open an issue, on it. 
@thibaudcolas  How about if I want to live preview the changes I make on particular page , let's say for example the about page in the /about-wagtail route?

**Reply by thibaudcolas** [reply]
@MugoBrian create that page in the CMS and view it in the browser?
